### PR TITLE
Staff with only manage_orders should be able to query all stock related date.

### PR DESCRIPTION
**Pull request by fowczarek**
I want to merge this change because staff with only manage_orders should be able to query all stock related date.

Staff member with `manage_orders` should be able to get information about warehouses and variant stocks.

```
    order(id: <OrderID>) {
        id
        lines {
            variant {
                stocks {
                    warehouse {
                        id
                    }
                    quantity
                    quantityAllocated
                }
            }
        }
    }
```

```
    warehouses(first:100, filter: $filters) {
        edges {
            node {
                ...
            }
        }
    }
```

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [x] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [x] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-1065 Order details page requires "manage products" permission](https://app.clickup.com/t/2549495/SALEOR-1065)

**Comment by patrys** [comment]
What information does that give them access to? I feel that we should explicitly list all fields and relations here (and in any similar pull requests where permissions are changed) and discuss whether any of them should be limited to one permission only.

**Comment by fowczarek** [comment]
@patrys  I add those fields to PR description.

**Comment by patrys** [comment]
@fowczarek I mean all the fields they get access to, not just the useful ones. I'm trying to avoid the situation where we accidentally expose a field or relation that the role should not get access to (one example could be the purchase cost of an item).

**Review by maarcingebala — APPROVED** [review]

File: saleor/graphql/order/tests/test_fulfillment.py [hunk 1]
```diff
@@ -1017,3 +1017,41 @@ def test_fulfillment_query(
...
+
+def test_staff_can_query_order_fulfill_data(
+    staff_api_client, order, permission_manage_orders
+):
+    order_id = graphene.Node.to_global_id("Order", order.pk)
+    variables = {"id": order_id}
+    response = staff_api_client.post_graphql(
+        QUERY_ORDER_FULFILL_DATA, variables, permissions=[permission_manage_orders]
+    )
+    _ = get_graphql_content(response)
```
Comment (tomaszszymanski129) [inline saleor/graphql/order/tests/test_fulfillment.py:1050]: could we check if the query actually return something instead of testing if it doesn't break?

**Review by tomaszszymanski129 — APPROVED** [review]

**Comment by maarcingebala** [comment]
@patrys From my perspective this PR looks like ready to merge, should we add or change anything here?

**Comment by patrys** [comment]
@maarcingebala I still don't know which fields become accessible as the result of this change so I can't tell whether we should limit access to some of them.

**Comment by maarcingebala** [comment]
Would like to have that in PR description / API description / somewhere else?

**Comment by patrys** [comment]
I don't have a preference as long as we can look at the list and say "this makes sense, this doesn't".

**Comment by maarcingebala** [comment]
| Field | Permission before | Permissions now|
|-------------------------|-------------------|--------------------------------------|
| `ProductVariant.stocks` | `MANAGE_PRODUCTS` | `MANAGE_PRODUCTS` or `MANAGE_ORDERS` |
| `warehouse`             | `MANAGE_PRODUCTS` | `MANAGE_PRODUCTS` or `MANAGE_ORDERS` |
| `warehouses`            | `MANAGE_PRODUCTS` | `MANAGE_PRODUCTS` or `MANAGE_ORDERS` |
| `Stock.quantity`            | `MANAGE_PRODUCTS` | `MANAGE_PRODUCTS` or `MANAGE_ORDERS` |
| `Stock.quantityAllocated`            | `MANAGE_PRODUCTS` | `MANAGE_PRODUCTS` or `MANAGE_ORDERS` |

**Review by patrys — APPROVED** [review]

**Pull request merged**